data-tck/src/main/java/io/micronaut/data/tck/repositories/JsonEntityRepository.java

```diff
@@ -0,0 +1,32 @@
+package io.micronaut.data.tck.repositories;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.data.annotation.Query;
+import io.micronaut.data.annotation.QueryResult;
+import io.micronaut.data.model.JsonType;
+import io.micronaut.data.repository.CrudRepository;
+import io.micronaut.data.tck.entities.JsonEntity;
+import io.micronaut.data.tck.entities.SampleData;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotNull;
+
+import java.util.Optional;
+
+public interface JsonEntityRepository extends CrudRepository<JsonEntity, Long> {
+
+    @Query("SELECT json_native AS DATA FROM json_entity WHERE id = :id")
+    @QueryResult(type = QueryResult.Type.JSON, jsonType = JsonType.NATIVE)
```

Comment:
Annotation can go on repository, but since this test has 3 different methods to covert different JSON representations when returning SampleData, don't know how could do it.

Reply:
We should support annotation `@JsonRepresentation` on the type just like we support `@TypeDef`

Reply:
But in this case `SampleData` is persisted in multiple ways so it wouldn't work.